src/aem/azext_aem/custom.py

```diff
@@ -0,0 +1,546 @@
+                    service_properties = storage_client.get_blob_service_properties()
+                    storage_cfg_ok = EnhancedMonitoring._check_storage_analytics(service_properties)
+                    if storage_cfg_ok:
+                        logger.warning('\t\tStorage Metrics configuration check: %s', succ_word)
+                    else:
+                        success = False
+                        logger.error('\t\tMetrics configuration check: %s', fail_word)
+                    if storage_cfg_ok:
+                        logger.warning("\t\tStorage Metrics data check for '%s'. Wait up to %s minutes ...",
+                                       storage_account_name, wait_time_in_minutes)
+                        filter_str = "Timestamp gt datetime'{}'".format(
+                            (datetime.datetime.utcnow() + datetime.timedelta(minutes=-5)).isoformat())
+                        result = self._check_table_and_content(storage_account_name, disk['key'],
+                                                               '$MetricsMinutePrimaryTransactionsBlob', filter_str,
+                                                               wait_time_in_minutes)
+                        if result:
+                            logger.warning("\t\tStorage Metrics data check '%s': %s", storage_account_name, succ_word)
+                        else:
+                            success = False
+                            logger.error("\t\tStorage Metrics data check '%s': %s", storage_account_name, fail_word)
+
+        logger.warning('Azure Enhanced Monitoring Extension for SAP public configuration check...')
+        extected, _ = self._build_extension_cfgs(disk_info)
+        public_cfg = {x['key']: x['value'] for x in self._vm.resources[0].settings['cfg']}
+        diffs = {k: [extected[k], public_cfg.get(k, None)] for k in extected if extected[k] != public_cfg.get(k, None)}
+        if diffs:
+            success = False
+            for err in diffs:
+                logger.error("\tConfiguration Error: Expected: '%s' Actual: '%s'", diffs[err][0], diffs[err][1])
+        else:
+            logger.warning('Configuration OK')
+
+        if not success:
+            raise CLIError('Configuration Not OK.')
+
+    def _build_extension_cfgs(self, disk_info):
+        vm_size = str(self._vm.hardware_profile.vm_size)
+        pub_cfg = pri_cfg = {}
+        vm_size_mapping = {
+            'ExtraSmall': 'ExtraSmall (A0)',
+            'Standard_A0': 'ExtraSmall (A0)',
+            'Basic_A0': 'ExtraSmall (A0)',
+            'Small': 'Small (A1)',
+            'Medium': 'Medium (A2)',
+            'Large': 'Large (A3)',
+            'ExtraLarge': 'ExtraLarge (A4)'
+        }
+        pub_cfg.update({
+            'vmsize': vm_size_mapping.get(vm_size, vm_size),
+            'vm.role': 'IaaS',
+            'vm.memory.isovercommitted': 0,
+            'vm.cpu.isovercommitted': 1 if vm_size == 'ExtraSmall (A0)' else 0,
```

Comment:
Do you also need to check for `vm_size == 'ExtraSmall'`?
Looking at the use of `vm_size_mapping.get(vm_size, vm_size)`, looks like you do need to check for that?